students/nDruP/lesson04/mailroom2.py

```diff
@@ -0,0 +1,167 @@
+              "Todd Bonzalez": [715867.83, 10352.07]}
+
+
+def main_menu():
+    """
+    Prompt user to send a Thank You, Create a report, create letters, or quit.
+    """
+    user_prompt = None
+    valid_prompts = {"1": craft_thank_u,
+                     "2": create_report,
+                     "3": create_letters,
+                     "4": sys.exit}
+    while True:
+        print_divider()
+        print("We're a Pyramid Scheme & So Are You! E-Mailroom")
+        print_divider()
+        while user_prompt not in valid_prompts:
+            print("Please choose from the following options (1,2,3,4): ")
+            print("1. Send a Thank you")
+            print("2. Create Donor Report")
+            print("3. Send letters to everyone")
+            print("4. Quit")
+            user_prompt = input(">")
+        print_divider()
+        valid_prompts.get(user_prompt)()
+        input("\nPress enter to continue...")
+        user_prompt = None
+
+
+def craft_thank_u():
+    """
+    Prompt for a full name.
+    Prompt->"list": show a list of the donor names
+    Prompt->name not in list: add name to list;
+    Then prompt for donation amount, add amount to donation history
+    Compose and print an email thanking the donor for their donation.
+    Return to main
+    """
+    donor_name = "list"
+    donation_amt = None
+    message = ''
+    while True:
+        print("Let's craft a very personal thank you note for our donor!")
+        print("Return to the main menu at any time by entering 'exit'")
+
+        print("Pull up a list of donor names by entering 'list'")
+        while donor_name == "list":
+            donor_name = input("Enter Donor Name >")
+            if donor_name.lower() == "list":
+                print(("{}\n"*len(donor_dict)).format(*donor_dict))
+        if donor_name.lower() == "exit":
+            break
+
+        donation_amt = input("Enter their Donation Amount >")
+        if donation_amt.lower() == "exit":
+            break
+        donation_amt = float(donation_amt)
+
+        if donor_name not in donor_dict:
+            donor_dict[donor_name] = []
+        donor_dict[donor_name].append(donation_amt)
+        print_divider()
+        message = f"Dearest {donor_name},\n"
+        message += f"Thank you so much for donation of ${donation_amt:.2f}!\n"
+        message += "We will use your donation to create a real living Pokemon."
+        message += f"\nSincerely,\nWe're a Pyramid Scheme & so is {donor_name}"
+        print(message)
+        print_divider()
+
+        break
+    return
+
+
+def create_report():
+    """
+    Print a list of donors sorted by total historical donation amount.
+    Donor Name, Total Given, Num Gifts, Average Gift
+    """
+    sort_donors = sorted(donor_dict.items(), key=sum_2tuple_by2, reverse=True)
+    col_name_list = ["Donor Name", "Total Given", "Gifts", "Avg Donation"]
+    billi_len = 12
+    donor_col_len = max([len(name) for name, donations in sort_donors])
```

Comment:
common pattern is to use `_` for var names for data you do not care about to indicate that clearly

for example:
```
 max([len(name) for name, _ in sort_donors])
```